x-pack/legacy/plugins/task_manager/task_manager.ts

```diff
@@ -219,6 +222,26 @@ export class TaskManager {
     return result;
   }
 
+  /**
+   * Schedules a task with an Id
+   *
+   * @param task - The task being scheduled.
+   * @returns {Promise<ConcreteTaskInstance>}
+   */
+  public async ensureScheduling(
+    taskInstance: ExistingTaskInstance,
+    options?: any
+  ): Promise<ExistingTaskInstance> {
```

Comment:
So I guess sometimes this will return a `ConcreteTaskInstance` (if not already scheduled), but otherwise the `taskInstance` sent in.  Seems like it would be "nice" if it did always return a `ConcreteTaskInstance`, but guessing you will have to do another read to get that, and hardly seems worth it, since the caller may not need it. Seems like this should be fine as a first attempt, maybe we'd need to re-look at a way to get the ConcreteTaskInstance back later, if we find we need it for some reason.

In any case, the JSDoc comment above this should be changed to say it's returning an `ExistingTaskInstance` instead of `ConcreteTaskInstance`.

Reply:
Yeah, I did think about that, though, the type it returns will always be `TaskInstanceWithId`, never an explicit `ConcreteTaskInstance`, as `TaskInstanceWithId` is essentially a subtype of `ConcreteTaskInstance`, but you're right that otherwise we'd have to make an additional read and it didn't seem worth it.
Feels reasonable to leave it such due to the _ensure_ nature, no?

I'll change the JSDoc, thanks for catching 👍